Examples/Python/python/acts/examples/detector.py

```diff
@@ -16,420 +17,169 @@
+        self._supports = supports
+        self._loglevel = loglevel
+
+    def builder(self):
+        "Return the associated builder"
+
+        # Get r, z, phi range
+        rRange = self._extent.range(acts.Binning.r)
+        zRange = self._extent.range(acts.Binning.z)
+
+        # Set up the shape builder: external builder
+        shapeConfig = VolumeStructureBuilder.Config()
+        shapeConfig.boundsType = VolumeBoundsType.Cylinder
+        shapeConfig.boundValues = [
+            rRange[0],
+            rRange[1],
+            0.5 * (zRange[1] - zRange[0]),
+            math.pi,
+            0,
+        ]
+        shapeConfig.transform = Transform3([0, 0, 0.5 * (zRange[1] + zRange[0])])
+        shapeConfig.auxiliary = "Shape[" + self._name + "]"
+
+        # Set up the volume builder
+        volConfig = acts.DetectorVolumeBuilder.Config()
+        volConfig.name = self._name
+        volConfig.auxiliary = "Volume[" + self._name + "]"
+        volConfig.externalsBuilder = VolumeStructureBuilder(
+            shapeConfig, shapeConfig.auxiliary, self._loglevel
         )
-
-        # Volume builder
-        volumeBuilderConfig = acts.DetectorVolumeBuilder.Config()
-        volumeBuilderConfig.name = self.__name
-        volumeBuilderConfig.internalsBuilder = layerStructureBuilder
-        volumeBuilderConfig.externalsBuilder = volumeStructureBuilder
-
-        volumeBuilder = acts.DetectorVolumeBuilder(
-            volumeBuilderConfig, self.__name + "_VolumeBuilder", acts.logging.VERBOSE
-        )
-
-        return volumeBuilder
-
-
-class ContainerStructure:
-    def __init__(self, name, volumes, binning):
-        """Creates a container from a given list of volumes
-
-        Parameters
-        ----------
-        name: string
-            The name of the container
-        volumes: list of DetectorVolume
-            The volumes of the container
-        binning: acts.Binning
-            The ordering of the container
-
-        Returns
-        -------
-        ContainerStructure
-            The created container proxy that will provide an appropriate builder.
-        """
-        self.__name = name
-        self.__volumes = volumes
-        self.__binning = binning
-
-    def getName(self):
-        return self.__name
-
-    def getBuilder(self):
-        """Returns the builder for the container"""
-        volumeBuilders = []
-        for volume in self.__volumes:
-            volumeBuilders += [volume.getBuilder()]
-
-        containerBuilder = None
-        if self.__binning == acts.Binning.r or self.__binning == acts.Binning.z:
-            containerConfig = acts.CylindricalContainerBuilder.Config()
-            containerConfig.builders = volumeBuilders
-
-            containerBuilder = acts.CylindricalContainerBuilder(
-                containerConfig, self.__name + "Builder", acts.logging.VERBOSE
+        if self._provider is not None:
+            layerConfig = LayerStructureBuilder.Config()
```

Comment:
Just for me to know, what does a layerbuilder do in the new geometry layout?

Reply:
It builds a detector volume that represents a layer.